exp/controllers/awsmachinepool_controller.go

```diff
@@ -493,36 +522,37 @@ func (r *AWSMachinePoolReconciler) findASG(machinePoolScope *scope.MachinePoolSc
 	return asg, nil
 }
 
-// asgNeedsUpdates compares incoming AWSMachinePool and compares against existing ASG.
-func asgNeedsUpdates(machinePoolScope *scope.MachinePoolScope, existingASG *expinfrav1.AutoScalingGroup) bool {
+// detectASGDrift compares incoming AWSMachinePool and compares against existing ASG.
+func detectASGDrift(machinePoolScope *scope.MachinePoolScope, existingASG *expinfrav1.AutoScalingGroup) string {
```

Comment:
nit: Let's please name the function after what it does, not the purpose for which it is used. For instance, `diffASG`.

Reply:
Done.